### ISSUE TITLE
Image Zooming Animation with onScroll Event

### ISSUE THREAD

**Issue by daesan** [facebook/react-native#6401]
I am trying to achieve Twitter Profile Page's Image Zooming (and Blurring) effect when the scroll view is pulled downward.

My current approach is to update _Animated.Value_ in _onScroll_ event callback to drive the zooming animation of the image at the top of _ScrollView_. The issue I am having with this approach is that the animation is quite flaky.

Does anybody have an idea to tackle this particular problem? Should I contemplate on implementing a custom native ScrollView with auto expanding component at the top?

Thanks for any feedback in advance!

![twitter_profile_image](https://cloud.githubusercontent.com/assets/2495/13666043/cc762e14-e6f2-11e5-945d-b48dcdff88c0.png)

**Comment by nicklockwood** [comment]
How are you implementing the blur effect?

**Comment by nicklockwood** [comment]
The zoom effect can be achieved by just setting the image's resizeMode to 'cover', then it will zoom automatically when the container is resized. That should be pretty smooth. There's no way to implement the blur without a custom native component though (unless there's already a 3rd party component to do this).

**Comment by daesan** [comment]
Thanks for the tip!

I did go on to implement a custom native ScrollView component to use "scrollViewDidScroll:" method to drive the zooming animation and I am quite happy with the result. I will try what you suggested and report back how both methods fare. It will be awesome if similar performance can be achieved without using native method!

For the blur effect, I first tried to wrap _UIVisualEffectView_ class and cover the header image with that custom view. However, apple's _UIVisualEffectView_ has a static blur effect. You cannot control the degree of blurriness in response to scrolling.

I am now thinking about implementing a custom native Image component with blurring effect built in. Not quite sure how to go about this yet.

**Comment by daesan** [comment]
I did try the method you suggested. While utilizing image's resizeMode makes the animation a lot smoother than zooming manually in JS, the lagging/jiggling is quite noticeable, especially when you pull down the scrollview and let go. For the zooming part, I think I will stick with custom scrollview approach.

For the blurring effect, I implemented custom native image view component and am using alpha attribute of the image view to change the blurriness dynamically with original image underneath. This way, I can simulate the blur effect without running blurring algorithm repeatedly which, I am afraid, may lag.
